### ISSUE TITLE
Signs only update on focus gain

### ISSUE THREAD

**Issue by leggsimon** [airblade/vim-gitgutter#298]
I'll admit, I'm pretty new to using Vim but I've installed this package today, and the `g:gitgutter_realtime` doesn't seem to be working. The signs only seems to update if I enter a new window and then focus on the original window again.

I've tried `set updatetime=250` and that doesn't appear to make a difference, if I run `:GitGutterEnable` it updates it straight away but I don't really want to run that everytime I make a change.

I currently have this in my `.vimrc`
```vim
" ~/.vimrc

let g:gitgutter_highlight_lines = 1
let g:gitgutter_realtime = 1
set updatetime=250
```

Is there anything I should be adding?

Also posted on StackOverflow [here](http://stackoverflow.com/questions/34845443/vim-gitgutter-signs-only-update-on-focus-gain)

**Comment by airblade** [comment]
Your configuration is fine (though you don't need to specify `let g:gitgutter_realtime=1` – it's the default).  With an update time of 250ms, the signs should update a quarter of a second after you stop typing.

So all the diff and sign machinery is working; it's a problem with the autocommands.  Please could you run `:au gitgutter` and post the output (a screenshot might be easiest)?

**Comment by leggsimon** [comment]
Yeah I thought not but I added it just incase it wasn't reading it.

Running `:au gitgutter` returns this
![au gitgutter](https://www.dropbox.com/s/kw66zox9p4r7d2u/Screenshot%202016-01-18%2010.53.09.png?dl=0)

**Comment by airblade** [comment]
Thanks.  You're missing the [`CursorHold` and `CursorHoldI` autocommands](https://github.com/airblade/vim-gitgutter/blob/ff4f592c582c0c4181f3c7e40977d1c00c392ce3/plugin/gitgutter.vim#L165-L167) which is why the realtime stuff isn't working.

You have already set `g:gitgutter_realtime` to `1`, which is the default anyway, so something else must be resetting your `CursorHold` and `CursorHoldI` autocommands.

Just to be sure, please could you fire up vim and confirm `:echo g:gitgutter_realtime` prints `1`?

If it does, something else in your vimrc or another plugin is resetting those autocommands.  What does `:au CursorHold` produce?

**Comment by leggsimon** [comment]
Hmm no, `:echo g:gitgutter_realtime` prints `0` for some reason

**Comment by airblade** [comment]
OK, your problem is that `g:gitgutter_realtime` is `0` not `1`.  It's `1` by default so somehow it is getting turned off.

The autocommand is `CursorHold` (with an `o` in cursor).

**Comment by leggsimon** [comment]
:facepalm: Oops

Interestingly, I have just opened up a new project with Vim and checked `:echo g:gitgutter_realtime` and it was `1`. So I have quit Terminal again and reopened the original project again and it appears to be working now. I had done that before but I wonder whether I added something to my `.vimrc` after restarting Terminal and forgot to do it again?

Having said that, I assume that if I `:q` out of Vim and go back into it it should've re-sourced my `.vimrc`?
